frontend/packages/console-shared/src/utils/pod-utils.ts

```diff
@@ -142,3 +146,35 @@ export const isIdled = (deploymentConfig: K8sResourceKind): boolean => {
     false,
   );
 };
+
+export const getPodData = (
+  dc: K8sResourceKind,
+  pods: Pod[],
+  current: PodControllerOverviewItem,
+  previous: PodControllerOverviewItem,
+  isRollingOut: boolean,
+): { inProgressDeploymentData: Pod[] | null; completedDeploymentData: Pod[] | null } => {
+  const strategy = _.get(dc, ['spec', 'strategy', 'type']);
```

Comment:
I used _.get here because for StatefulSets and daemonSets `strategy` doesn't exist. For straight accessing, I would need to apply check for spec then the strategy and then get the type.

Reply:
Fair enough - use the 3rd param (even if you're setting it to `null`) and type `strategy`, we have TypeScript so we should leverage readability through types.

Reply:
fixed in https://github.com/openshift/console/pull/2496/commits/6353ee79b77029607673e09df07a9099b0fb4a9b